test/apple_shell_testutils.sh

```diff
@@ -523,21 +523,19 @@ function is_device_build() {
 }
 
 
-# Usage: print_debug_entitlements <binary_path>
+# Usage: print_debug_entitlements <binary_path> <output>
 #
-# Extracts and prints the debug entitlements from the appropriate Mach-O
-# section of the given binary.
+# Extracts and prints the debug entitlements from the the given binary
+# using segedit.
 function print_debug_entitlements() {
   local binary="$1"
-
-  # This monstrosity uses objdump to dump the hex content of the entitlements
-  # section, strips off the leading addresses (and ignores lines that don't
-  # look like hex), then runs it through `xxd` to turn the hex into ASCII.
-  # The results should be the entitlements plist text, which we can compare
-  # against.
-  xcrun llvm-objdump --macho --section=__TEXT,__entitlements "$binary" | \
-      sed -e 's/^[0-9a-f][0-9a-f]*[[:space:]][[:space:]]*//' \
-          -e 'tx' -e 'd' -e ':x' | xxd -r -p
+  local output="$2"
+  local archs=($(lipo -archs "$binary"))
+  local thin_binary="tempdir/thin_binary"
+  mkdir -p tempdir
+  lipo -thin ${archs[0]} "$binary" -output "$thin_binary"
+  xcrun segedit "$thin_binary" -extract __TEXT __entitlements "$output"
```

Comment:
couldn't we output this to `-` as stdout instead?

Reply:
Happy to take suggestions though!

Reply:
I was suggesting to do the lipo, and then just make the segedit part output to stdout so you don't have to make callers know about the temporary files

Reply:
I'll merge and investigate